user-facing-messages.md

```diff
@@ -0,0 +1,432 @@
+INFO:     Started server process [81728]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: Salesforce REST API version '52.1' is not supported. Update 'salesforce-api-version' in project.toml to '53.0' or newer.
+
+ERROR:    Application startup failed. Exiting.
+```
+
+### tests/fixtures/project_toml_api_version_triple_digits
+
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: A main.py file was not found at: /Users/emorley/src/sf-functions-python/tests/fixtures/project_toml_api_version_triple_digits/main.py
+```
+
+In the context of the CLI start command:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81730]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: A main.py file was not found at: /Users/emorley/src/sf-functions-python/tests/fixtures/project_toml_api_version_triple_digits/main.py
+
+ERROR:    Application startup failed. Exiting.
+```
+
+### tests/fixtures/project_toml_api_version_wrong_type
+
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: The 'com.salesforce.salesforce-api-version' key in project.toml must be a string.
+```
+
+In the context of the CLI start command:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81732]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: The 'com.salesforce.salesforce-api-version' key in project.toml must be a string.
+
+ERROR:    Application startup failed. Exiting.
+```
+
+### tests/fixtures/project_toml_file_missing
+
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: A project.toml file was not found at: /Users/emorley/src/sf-functions-python/tests/fixtures/project_toml_file_missing/project.toml
+```
+
+In the context of the CLI start command:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81734]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: A project.toml file was not found at: /Users/emorley/src/sf-functions-python/tests/fixtures/project_toml_file_missing/project.toml
+
+ERROR:    Application startup failed. Exiting.
+```
+
+### tests/fixtures/project_toml_invalid_toml
+
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: project.toml is not valid TOML: Expected '=' after a key in a key/value pair (at line 1, column 4)
+```
+
+In the context of the CLI start command:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81736]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: project.toml is not valid TOML: Expected '=' after a key in a key/value pair (at line 1, column 4)
```

Comment:
For this and the next one, the changes to the start of the error message look great, however I'm less sure about the trailing full stop at the end, only because we don't generate the error message that's appended - so the library might include some that already have a trailing full stop, which would mean a double full stop.

Reply:
makes sense!  so ignore it.  thx for explaining :)